src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1521,16 +1556,26 @@ public static String verifyVirtualMachineImage(
     public static String verifyStorageAccountName(
             final ServicePrincipal servicePrincipal,
             final String resourceGroupName,
-            final String storageAccountName) {
+            final String storageAccountName,
+            final String storageAccountType) {
         boolean isAvailable = false;
         try {
             Azure azureClient = TokenCache.getInstance(servicePrincipal).getAzureClient();
             //if it's not available we need to check if it's already in our resource group
             isAvailable = azureClient.storageAccounts().checkNameAvailability(storageAccountName).isAvailable();
-            if ( !isAvailable && null == azureClient.storageAccounts().getByGroup(resourceGroupName, storageAccountName) ) {
+            StorageAccount checkAccount = azureClient.storageAccounts().getByGroup(resourceGroupName, storageAccountName);
```

Comment:
when I was trying this feature, I met one issue is if the account name is not valid (for example, too long) it will check availability will also return false but we will treat it as already exists, which is misleading. Could you add another check to check whether the name is valid first?